app/frontend/wallet/account.ts

```diff
@@ -500,6 +500,7 @@ const Account = ({
     getPoolRecommendation,
     isAccountUsed,
     ensureXpubIsExported,
+    _getKeys: getKeys,
```

Comment:
I checked the `getKeys()` implementation:
```
async function getKeys() {
    const shelleyAccountXpub = await getAccoutXpubShelley(cryptoProvider, accountIndex)
    const byronAccountXpub = await getAccoutXpubByron(cryptoProvider, accountIndex)
    const stakingKeyCborHex = await getStakingKeyCborHex(cryptoProvider, accountIndex)
    const stakingAddress = await myAddresses.accountAddrManager._deriveAddress(accountIndex)
    const stakingAddressHex = await getStakingAddressHex(cryptoProvider, accountIndex)
    return {
      shelleyAccountXpub,
      byronAccountXpub,
      stakingKeyCborHex,
      stakingAddress,
      stakingAddressHex,
    }
  }
```

 and it seems to me that it should be named differently or better, broken down into e.g. `getAccountKeys()` and `getStakingCredentials()`. And `stakingKeyCborHex` would probably be neater if it was just the hex of the public key (possibly even xpub) and do the cbor-encoding wherever needed with some simple utility function (e.g. xpubToCborHex) or something like that as it is a matter of presentation and is preferable to hold data in the state in a standardized format and not bloat it with different presentations of the same piece of information which can easily get out of sync when implementing changes. Same goes for `stakingAddressHex` as the hex representation of the address could easily be extracted with a helper function bech32ToHex wherever needed, avoiding duplication and bloating in the state.

Reply:
You are right, I left this half-refactored and targeted mostly on fixing the names of the variables and I united keys and addresses into one function, but I ll fix it, and regarding the state pollution, the reason I put multiple versions of the same key, like hex and cbor into state was to separate the encoding logic from the `keys.tsx` component, but you are right, we could create the mentioned helper functions and do it that way. Thanks.